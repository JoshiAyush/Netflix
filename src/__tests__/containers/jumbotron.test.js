import React from "react";
import { render } from "@testing-library/react";

import jumboData from "../../fixtures/jumbo.json";

import { JumbotronContainer } from "../../containers/index.js";

describe("<Jumbotron />", () => {
  it("renders the <Jumbotron /> with populated data", () => {
    const { container, getByText, getByAltText, getByTestId } = render(
      <JumbotronContainer.Container>
        {jumboData.map((item) => (
          <JumbotronContainer key={item.id} direction={item.direction}>
            <JumbotronContainer.Pane>
              <JumbotronContainer.Title>{item.title}</JumbotronContainer.Title>

              <JumbotronContainer.SubTitle>
                {item.subTitle}
              </JumbotronContainer.SubTitle>
            </JumbotronContainer.Pane>

            <JumbotronContainer.Pane>
              <JumbotronContainer.Image
                src={item.image}
                alt={item.alt}
                data-testid={`${item.id}--jumbo--image`}
              />
            </JumbotronContainer.Pane>
          </JumbotronContainer>
        ))}
      </JumbotronContainer.Container>
    );

    expect(getByText("Enjoy on your TV.")).toBeTruthy();
    expect(getByAltText("Tiger King on Netflix")).toBeTruthy();
    expect(getByTestId("1--jumbo--image")).toBeTruthy();
    expect(
      getByText(
        "Watch on smart TVs, PlayStation, Xbox, Chromecast, Apple TV, Blu-ray players and more."
      )
    ).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
