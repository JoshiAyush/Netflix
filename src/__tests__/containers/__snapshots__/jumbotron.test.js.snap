// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Jumbotron /> renders the <Jumbotron /> with populated data 1`] = `
<div
  class="style__Container-mps17o-2 fBieZc"
>
  <div
    class="style__Item-mps17o-0 izMvgc"
  >
    <div
      class="style__Inner-mps17o-1 bIrjoR"
      direction="row"
    >
      <div
        class="style__Pane-mps17o-3 kKhzWn"
      >
        <h1
          class="style__Title-mps17o-4"
        >
          Enjoy on your TV.
        </h1>
        <h2
          class="style__SubTitle-mps17o-5 gOufQF"
        >
          Watch on smart TVs, PlayStation, Xbox, Chromecast, Apple TV, Blu-ray players and more.
        </h2>
      </div>
      <div
        class="style__Pane-mps17o-3 kKhzWn"
      >
        <img
          alt="Tiger King on Netflix"
          class="style__Image-mps17o-6 gvxiGB"
          data-testid="1--jumbo--image"
          src="/images/misc/home-tv.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="style__Item-mps17o-0 izMvgc"
  >
    <div
      class="style__Inner-mps17o-1 kzIHxc"
      direction="row-reverse"
    >
      <div
        class="style__Pane-mps17o-3 kKhzWn"
      >
        <h1
          class="style__Title-mps17o-4"
        >
          Download your programmes to watch on the go.
        </h1>
        <h2
          class="style__SubTitle-mps17o-5 gOufQF"
        >
          Save your data and watch all your favourites offline.
        </h2>
      </div>
      <div
        class="style__Pane-mps17o-3 kKhzWn"
      >
        <img
          alt="Watch on mobile"
          class="style__Image-mps17o-6 gvxiGB"
          data-testid="2--jumbo--image"
          src="/images/misc/home-mobile.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="style__Item-mps17o-0 izMvgc"
  >
    <div
      class="style__Inner-mps17o-1 bIrjoR"
      direction="row"
    >
      <div
        class="style__Pane-mps17o-3 kKhzWn"
      >
        <h1
          class="style__Title-mps17o-4"
        >
          Watch everywhere.
        </h1>
        <h2
          class="style__SubTitle-mps17o-5 gOufQF"
        >
          Stream unlimited films and TV programmes on your phone, tablet, laptop and TV without paying more.
        </h2>
      </div>
      <div
        class="style__Pane-mps17o-3 kKhzWn"
      >
        <img
          alt="Money Heist on Netflix"
          class="style__Image-mps17o-6 gvxiGB"
          data-testid="3--jumbo--image"
          src="/images/misc/home-imac.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
